File: rust/azure_iot_operations_protocol/tests/metl/default_prologue.rs
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.

use serde::Deserialize;

use crate::metl::default_executor::DefaultExecutor;
use crate::metl::default_invoker::DefaultInvoker;
use crate::metl::default_sender::DefaultSender;

#[derive(Deserialize, Debug)]
pub struct DefaultPrologue {
    #[serde(rename = "executor")]
    pub executor: Option<DefaultExecutor>,

    #[serde(rename = "invoker")]
    pub invoker: Option<DefaultInvoker>,

    #[serde(rename = "sender")]
    pub sender: Option<DefaultSender>,
}

File: go/test/protocol/default_prologue.go
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT License.
package protocol

type DefaultPrologue struct {
	Executor DefaultExecutor `toml:"executor"`
	Invoker  DefaultInvoker  `toml:"invoker"`
	Receiver DefaultReceiver `toml:"receiver"`
	Sender   DefaultSender   `toml:"sender"`
}
